### ISSUE TITLE
Templated Inputs

### ISSUE THREAD

**Issue by dglazkov** [breadboard-ai/breadboard#3260]
Writing this idea down for later:

Imagine a component that has a "input template" config port. 

You type into this port:

```
Hello! Welcome to zombo.com. 

Let's start by getting a few details about your business:

Name: {{business}}
Location: {{location}}
A picture of your business:
{{picture | image}}
```
When invoked, this component will show the LLM Input field of the following structure:

- Part 1 text (read-only): "Hello! Welcome to zombo ... etc etc ... Name:"
- Part 2 text: a place for the user to enter the business name
- Part 3 text (read-only): "Location:"
- Part 3 text: a place for the user to enter the location
- Part 4 text (read-only): "A picture of your business: "
- Part 5 image: a place to upload an image or take a webcam picture

So now, it looks like we presented the user with a form. The user fills out the fields they can edit and hit "Continue".

There's another optional config port named "output template" that allows you to format the output:

```
Name: {{business}}
Location: {{location}}
Picture: {{picture}}
```

The output port is "Context Out", and for our case, it will contain two parts:

- Part 1 text: "Name: etc etc." everything until "Picture:"
- Part 2 image: the picture.

If the output template is not specified, the input template is used.

**Comment by dglazkov** [comment]
@paullewis

**Comment by paullewis** [comment]
I think we could make this out of the existing LLM Content type. The only bits of metadata we'd need to add in are a) the type of each segment so that only offer the right input controls, and b) which ones are read-only versus editable. Besides that, though, I think we're already pretty near.

**Comment by dglazkov** [comment]
Yeah! I was thinking it will just react to the LLM content it's given. 
If a part has a `readonly` property, then it's frozen. 
Re: the right input controls. Maybe specify schema on a non-read-only part? Similar to my hacks with $metadata, but on each part?

**Comment by paullewis** [comment]
Guessing we don't need this nowadays?
